### ISSUE TITLE
sssd-kcm crashes with multiple parallel requests

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#3458]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/3470
- Created at 2017-08-10 13:28:36 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2019-03-20 23:01:27 as worksforme
- Assigned to nobody
---

Ticket was cloned from Red Hat Bugzilla (product _Red Hat Enterprise Linux 7_): [Bug 1465457](https://bugzilla.redhat.com/show_bug.cgi?id=1465457)

```text
Description of problem:

sssd_kcm crashes when kinit is run by multiple users repetitively (in a loop).

Version-Release number of selected component (if applicable):


sssd-client-1.15.2-50.el7.x86_64
sssd-ipa-1.15.2-50.el7.x86_64
sssd-proxy-1.15.2-50.el7.x86_64
python-sssdconfig-1.15.2-50.el7.noarch
sssd-common-1.15.2-50.el7.x86_64
sssd-common-pac-1.15.2-50.el7.x86_64
sssd-ad-1.15.2-50.el7.x86_64
sssd-ldap-1.15.2-50.el7.x86_64
sssd-1.15.2-50.el7.x86_64
sssd-kcm-1.15.2-50.el7.x86_64
sssd-krb5-common-1.15.2-50.el7.x86_64
sssd-krb5-1.15.2-50.el7.x86_64
libtevent-0.9.31-1.el7.x86_64
libtalloc-2.1.9-1.el7.x86_64

How reproducible:


Steps to Reproduce:
1. Join RHEL4 system to Windows AD using realm

2. Enable sssd-kcm

3.
sssd.conf
[sssd]
config_file_version = 2
services = nss, pam
domains = centaur.test

[domain/centaur.test]
ad_domain = centaur.test
krb5_realm = CENTAUR.TEST
cache_credentials = True
id_provider = ad
krb5_store_password_if_offline = True
default_shell = /bin/bash
ldap_id_mapping = True
use_fully_qualified_names = True
fallback_homedir = /home/%u@%d
ldap_id_mapping = True
access_provider = ad
enumerate = True
debug_level = 6

[pam]
debug_level = 9

[ssh]

[sudo]

[kcm]
debug_level = 9



4. Create 3 users in AD (foobar1, foobar2, foobar3 with password Secret123)

Open 3 terminals and run the below in parallel

#/bin/bash
export KRB5CCNAME=KCM:
date
echo Secret123 | kinit foobar1
date;for i in $(seq 1 100); do klist -A; done

On terminal 2

export KRB5CCNAME=KCM:
date
echo Secret123 | kinit foobar2
date;for i in $(seq 1 100); do klist -A; done

On terminal 3 run,

export KRB5CCNAME=KCM:
date
echo Secret123 | kinit foobar3
date;for i in $(seq 1 100); do klist -A; done


Actual results:
sssd-kcm crashes.


Expected results:
sssd-kcm should not crash.


Additional info:
Refer: https://bugzilla.redhat.com/show_bug.cgi?id=1446302#c11
```

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-08-10 13:28:38](https://pagure.io/SSSD/sssd/issue/3470#456594)**

**Metadata Update from @jhrozek**:
- Custom field rhbz adjusted to https://bugzilla.redhat.com/show_bug.cgi?id=1465457

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-08-10 13:28:38](https://pagure.io/SSSD/sssd/issue/3470#456595)**

**Metadata Update from @jhrozek**:
- Custom field rhbz adjusted to https://bugzilla.redhat.com/show_bug.cgi?id=1465457

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-08-10 13:28:56](https://pagure.io/SSSD/sssd/issue/3470#456596)**

**Metadata Update from @jhrozek**:
- Issue set to the milestone: SSSD 1.15.4 (was: SSSD 1.15 backlog)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-08-11 13:20:23](https://pagure.io/SSSD/sssd/issue/3470#457081)**

Check downstream comments https://bugzilla.redhat.com/show_bug.cgi?id=1446302#c11
and https://bugzilla.redhat.com/show_bug.cgi?id=1446302#c26

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-08-18 16:11:09](https://pagure.io/SSSD/sssd/issue/3470#458729)**

**Metadata Update from @jhrozek**:
- Issue priority set to: major

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-10-19 21:05:34](https://pagure.io/SSSD/sssd/issue/3470#473687)**

**Metadata Update from @jhrozek**:
- Issue set to the milestone: SSSD 1.16.1 (was: SSSD 1.15.4)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-11-15 21:29:59](https://pagure.io/SSSD/sssd/issue/3470#479564)**

I tend to do a lot of ssh calls in a loop and such, which can provoke this bug rather easily.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-11-16 08:53:03](https://pagure.io/SSSD/sssd/issue/3470#479642)**

**Metadata Update from @lslebodn**:
- Issue tagged with: KCM

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-12-15 19:45:28](https://pagure.io/SSSD/sssd/issue/3470#484924)**

**Metadata Update from @jhrozek**:
- Issue tagged with: bug, postpone-to-1-16-2

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-01-08 17:27:51](https://pagure.io/SSSD/sssd/issue/3470#487176)**

**Metadata Update from @jhrozek**:
- Issue set to the milestone: SSSD 1.16.2 (was: SSSD 1.16.1)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-01-08 17:39:42](https://pagure.io/SSSD/sssd/issue/3470#487200)**

**Metadata Update from @jhrozek**:
- Issue **un**tagged with: postpone-to-1-16-2

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-04-27 13:57:36](https://pagure.io/SSSD/sssd/issue/3470#508940)**

**Metadata Update from @fidencio**:
- Issue close_status updated to: Fixed
- Issue status updated to: Closed (was: Open)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-04-27 13:58:47](https://pagure.io/SSSD/sssd/issue/3470#508941)**

Sorry, wrongly closed this PR. :-/

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-04-27 13:58:47](https://pagure.io/SSSD/sssd/issue/3470#508942)**

**Metadata Update from @fidencio**:
- Issue status updated to: Open (was: Closed)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-06-05 15:08:38](https://pagure.io/SSSD/sssd/issue/3470#515353)**

Since we are near the 1.16.2 release and this ticket has no PR yet, it will slip into 1.16.3.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-06-05 15:08:42](https://pagure.io/SSSD/sssd/issue/3470#515354)**

**Metadata Update from @jhrozek**:
- Issue set to the milestone: SSSD 1.16.3 (was: SSSD 1.16.2)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-06-21 15:33:32](https://pagure.io/SSSD/sssd/issue/3470#517918)**

**Metadata Update from @fidencio**:
- Issue assigned to fidencio

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-06-21 15:41:49](https://pagure.io/SSSD/sssd/issue/3470#517922)**

**Metadata Update from @fidencio**:
- Assignee reset

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-08-10 11:56:42](https://pagure.io/SSSD/sssd/issue/3470#525551)**

**Metadata Update from @jhrozek**:
- Issue set to the milestone: SSSD 1.16.4 (was: SSSD 1.16.3)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2019-03-20 23:01:28](https://pagure.io/SSSD/sssd/issue/3470#561824)**

We could not reproduce this with a recent version, closing

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2019-03-20 23:01:28](https://pagure.io/SSSD/sssd/issue/3470#561825)**

**Metadata Update from @jhrozek**:
- Issue close_status updated to: worksforme
- Issue status updated to: Closed (was: Open)
